packages/eslint-plugin-baseui/src/deprecated-theme-api.js

```diff
@@ -0,0 +1,434 @@
+  },
+  font550: {
+    concern: 'typography',
+    replacement: 'HeadingXSmall',
+  },
+  font650: {
+    concern: 'typography',
+    replacement: 'HeadingSmall',
+  },
+  font750: {
+    concern: 'typography',
+    replacement: 'HeadingMedium',
+  },
+  font850: {
+    concern: 'typography',
+    replacement: 'HeadingLarge',
+  },
+  font950: {
+    concern: 'typography',
+    replacement: 'HeadingXLarge',
+  },
+  font1050: {
+    concern: 'typography',
+    replacement: 'HeadingXXLarge',
+  },
+  font1150: {
+    concern: 'typography',
+    replacement: 'DisplayXSmall',
+  },
+  font1250: {
+    concern: 'typography',
+    replacement: 'DisplaySmall',
+  },
+  font1350: {
+    concern: 'typography',
+    replacement: 'DisplayMedium',
+  },
+  font1450: {
+    concern: 'typography',
+    replacement: 'DisplayLarge',
+  },
+};
+
+module.exports = {
+  meta: {
+    fixable: 'code',
+    messages: {
+      [MESSAGES.deprecateThemeProperty.id]:
+        MESSAGES.deprecateThemeProperty.message,
+      [MESSAGES.replaceThemeProperty.id]: MESSAGES.replaceThemeProperty.message,
+    },
+  },
+  create(context) {
+    return {
+      Identifier(node) {
+        const identifier = node.name;
+
+        // We use `hasOwnProperty`, otherwise we might get false positives on the prototype chain.
+        if (
+          Object.prototype.hasOwnProperty.call(
+            deprecatedThemeProperties,
+            identifier,
+          )
+        ) {
+          // We have found an identifier that matches one of our deprecated theme properties.
+          const sourceCode = context.getSourceCode();
+          const scope = context.getScope(node);
+          const data = {
+            old: identifier,
+          };
+          let fix = null;
+          let messageId = MESSAGES.deprecateThemeProperty.id;
+
+          // If there is a possible replacement for this deprecated property,
+          // add a fix to replace with a new semantic property.
+          // TODO: use `suggest` instead, currently not supported in VSCode extension:
+          // https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions
+          // https://github.com/microsoft/vscode-eslint/issues/806
+          if (deprecatedThemeProperties[identifier].replacement) {
+            messageId = MESSAGES.replaceThemeProperty.id;
+            data.new = deprecatedThemeProperties[identifier].replacement;
+            fix = function(fixer) {
+              return fixer.replaceText(
+                node,
+                deprecatedThemeProperties[identifier].replacement,
+              );
+            };
+          }
+
+          // Check if we are setting a property in createTheme:overrides
+          // Ex: createTheme({}, {datepickerBackground: "pink"});
+          if (
+            context
+              .getAncestors()
+              .some(
+                node =>
+                  node.type === 'CallExpression' &&
+                  node.callee.name === 'createTheme',
```

Comment:
is it possible to check that `createTheme` was imported from `baseui`?

Reply:
Yeah, I started checking for imports in `deprecated-component-api` so I can use the same pattern here.

Reply:
Added a check for import here.